Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,176 @@
+        }
+
+        public static void PromptUserForStats()
+        {
+
+            string name = PromptNameInput("What is your name?");
+            int age = PromptAgeInput("How old are you? If I may ask");
+            float weight = PromptFloatInput("What is your weight? In Kilograms please.");
+            float height = PromptFloatInput("How tall are you? In Meters please.");
+
+            bool isHeightAndWeightValid = ValidateHeightAndWeight(weight, height);
+
+            if (isHeightAndWeightValid)
+            {
+                ProcessUserStats(name, age, weight, height);
+            }
+            else
+            {
+                Console.WriteLine("The values you provided were not valid.");
+            }
+
+        }
+
+        public static void ProcessUserStats(string name, int age, float weight, float height)
+        {
+            float calculatedBMI = CalculateBmi(weight, height);
+            Console.WriteLine($"{name} is {age} years old, weight is {weight} kg with a height of {height} meters.");
+            Console.WriteLine($"The calculated BMI  is: {calculatedBMI:F}");
+        }
+
+        public static string PromptNameInput(string promptMessage)
+        {
+            Console.WriteLine(promptMessage);
+
+            var name = Console.ReadLine();
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                Console.Write("Name cannot be empty.");
+                return "-";
+            }
+            else
+            {
+                return name;
+            }
+        }
+        
+        public static int PromptAgeInput(string promptMessage)
+        {
+            Console.WriteLine(promptMessage);
+            string age = Console.ReadLine();
+
+            if (string.IsNullOrEmpty(age))
+            {
+                return 0;
+            }
+            else if (IsValidPositiveNumber(age))
+            {
+                return int.Parse(age);             
+            }
+            else
+            {
+                DisplayInvalidInputErrorMessage(age);
+                return -1;
+            }
+        }
+
+        internal static bool IsValidPositiveNumber(string value) => IsNumber(value) && IsPositive(value);
+
+        public static float PromptFloatInput(string promptMessage)
+        {
+            Console.WriteLine(promptMessage);
+
+            var value = Console.ReadLine();
+            if (IsNumber(value) && IsPositive(value))
+            {
+                return float.Parse(value);
+            }
+            else if(string.IsNullOrEmpty(value))
+            {
+                return 0;
+            }
+            else
+            {
+                DisplayInvalidInputErrorMessage(value);
+                return -1;
+            }
+        }
+       
+        public static bool ValidateHeightAndWeight(float weight, float height)
+        {
+            bool isValidWeight = weight > 0, isValidHeight = height > 0;
```

Comment:
While this is short and compact I dislike this for several reasons. As those might be vey subjective I hold back on them, though. Just wanted to point this out! :)